File: Lecture034 Recursion Day4/bubbleSort.cpp
#include<iostream>
using namespace std;

void sortArray(int *arr, int n) {

    //base case - already sorted
    if(n == 0 || n == 1) {
        return ;
    }

    //1 case sovle karlia - largest element ko end me rakh dega
    for(int i=0; i<n-1; i++) {
        if(arr[i] > arr[i+1]){
            swap(arr[i], arr[i+1]);
        }
    }

    //Recursive Call
    sortArray(arr, n-1);

}

int main() {

    int arr[5] = {2,5,1,6,9};
    
    sortArray(arr,5);

    for(int i=0; i<5; i++) {
        cout << arr[i] << " ";
    }cout << endl;


    return 0;
}

File: Lecture034 Recursion Day4/BubbleSort.java
public class BubbleSort {
    public static void main(String[] args) {
        //Array to be sorted
        int[] arr = {2 , 8 , 4 , 9 , 5};
        //size of array
        int n = 5;

            //printing array before sorting
        System.out.println("Before");

        for(int i : arr){
            System.out.print(i + " ");
        }

        System.out.println();

        //recursive call
        sortRecursive(arr , n);

            //print array after sorting
        System.out.println("After");

        for(int i : arr){
            System.out.print(i + " ");
        }
    }
    public static void sortRecursive(int[] arr , int n){
        //base case
        if(n == 0 || n == 1)
            return;

        for(int i = 0; i < n - 1; i++){
            if(arr[i] > arr[i + 1]){
                int temp = arr[i];
                arr[i] = arr[i + 1];
                arr[i + 1] = temp;
            }
        }
         sortRecursive(arr , n - 1);
    }
}
